x-pack/legacy/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -160,39 +163,30 @@ export const elasticsearchMonitorsAdapter: UMMonitorsAdapter = {
     return monitorChartsData;
   },
 
-  getFilterBar: async ({ callES, dateRangeStart, dateRangeEnd }) => {
-    const fields: { [key: string]: string } = {
-      ids: 'monitor.id',
-      schemes: 'monitor.type',
-      urls: 'url.full',
-      ports: 'url.port',
-      locations: 'observer.geo.name',
-    };
+  getFilterBar: async ({ callES, dateRangeStart, dateRangeEnd, search, filterOptions }) => {
+    const aggs = generateFilterAggs(
+      [
+        { aggName: 'locations', filterName: 'locations', field: 'observer.geo.name' },
+        { aggName: 'ports', filterName: 'ports', field: 'url.port' },
+        { aggName: 'schemes', filterName: 'schemes', field: 'monitor.type' },
+        { aggName: 'tags', filterName: 'tags', field: 'tags' },
+      ],
+      filterOptions
+    );
+    const filters = combineRangeWithFilters(dateRangeStart, dateRangeEnd, search);
     const params = {
       index: INDEX_NAMES.HEARTBEAT,
       body: {
         size: 0,
         query: {
-          range: {
-            '@timestamp': {
-              gte: dateRangeStart,
-              lte: dateRangeEnd,
-            },
-          },
+          ...filters,
         },
-        aggs: Object.values(fields).reduce((acc: { [key: string]: any }, field) => {
-          acc[field] = { terms: { field, size: 20 } };
-          return acc;
-        }, {}),
+        aggs,
       },
     };
-    const { aggregations } = await callES('search', params);
 
-    return Object.keys(fields).reduce((acc: { [key: string]: any[] }, field) => {
-      const bucketName = fields[field];
-      acc[field] = aggregations[bucketName].buckets.map((b: { key: string | number }) => b.key);
-      return acc;
-    }, {});
+    const { aggregations } = await callES('search', params);
+    return extractFilterAggsResults(aggregations, ['tags', 'locations', 'ports', 'schemes']);
```

Comment:
This file seems to be broken up in a way that's no longer sustainable. It looks like we're trying to move code out to separate files (like `extract_filter_aggs.ts`) here, to keep the file from getting absurdly long.

It would be much cleaner to have one file with the backend logic for this API endpoint (including the `extractFilterAggsResults` function in it. In other words, we should have one file per endpoint here, instead of one big file with lots of helpers IMHO.

Reply:
I'm ++++ on breaking up these adapter files, but I'd prefer to do it in a separate PR. I've created a [tech debt issue](https://github.com/elastic/uptime/issues/131) for this work, we should resolve soon.

In the meantime, I have moved `extractFilterAggsResults` back into this file https://github.com/elastic/kibana/pull/50837/commits/967adf54ecf261a28bcb46ec55329af966ca7621.